src/backend/logging/log_manager.h

```diff
@@ -84,40 +122,110 @@ class LogManager {
+  BackendLogger *GetBackendLogger(unsigned int hint = 0);
 
+  // set the name of the log file (only used in wbl)
   void SetLogFileName(std::string log_file);
 
+  // get the log file name (used only in wbl)
   std::string GetLogFileName(void);
 
   bool HasPelotonFrontendLogger() const {
-    return (peloton_logging_mode == LOGGING_TYPE_NVM_NVM);
+    return (peloton_logging_mode == LOGGING_TYPE_NVM_WAL);
   }
 
+  // Drop all default tiles for tables before recovery
+  void PrepareRecovery();
+
+  // Add default tiles for tables if necessary
+  void DoneRecovery();
+
   //===--------------------------------------------------------------------===//
   // Utility Functions
   //===--------------------------------------------------------------------===//
 
-  FrontendLogger *GetFrontendLogger();
+  // initialize a list of frontend loggers
+  void InitFrontendLoggers();
+
+  // get a frontend logger at given index
+  FrontendLogger *GetFrontendLogger(unsigned int logger_idx);
 
-  void ResetFrontendLogger();
+  // remove all frontend loggers (used for testing)
+  void DropFrontendLoggers();
 
+  // perpare to log must be called before a commit id is generated for a transaction
+  void PrepareLogging();
+
+  // log the beginning of a commited transaction
   void LogBeginTransaction(cid_t commit_id);
 
-  void LogUpdate(concurrency::Transaction *curr_txn, cid_t commit_id,
-                 ItemPointer &old_version, ItemPointer &new_version);
+  // log an update
+  void LogUpdate(cid_t commit_id, ItemPointer &old_version,
+                 ItemPointer &new_version);
 
-  void LogInsert(concurrency::Transaction *curr_txn, cid_t commit_id,
-                 ItemPointer &new_location);
+  // log an insert
+  void LogInsert(cid_t commit_id, ItemPointer &new_location);
 
+  // log a delete
   void LogDelete(cid_t commit_id, ItemPointer &delete_location);
 
+  // commit a transaction and wait until stable
   void LogCommitTransaction(cid_t commit_id);
 
+  // used by the checkpointer to truncate unneeded log files
+  void TruncateLogs(txn_id_t commit_id);
+
+  // called if a transaction aborts before starting a commit
+  void DoneLogging();
+
+  // the maximum flushed commit id for all frontend loggers
+  cid_t GetGlobalMaxFlushedIdForRecovery() {
+    return global_max_flushed_id_for_recovery;
+  }
+
+  // set the max flushed id for recovery
+  void SetGlobalMaxFlushedIdForRecovery(cid_t new_max) {
+    global_max_flushed_id_for_recovery = new_max;
+  }
+
+  // updates the catalog and transaction managers to the correct oid and cid after recovery
+  void UpdateCatalogAndTxnManagers(oid_t new_oid, cid_t new_cid);
+
+  // set the maximum commit id which has been persisted to disk
+  void SetGlobalMaxFlushedCommitId(cid_t);
+
+  // set the maximum commit id which has been persisted to disk
+  cid_t GetGlobalMaxFlushedCommitId();
+
+  // get the list of frontend loggers
+  std::vector<std::unique_ptr<FrontendLogger>> &GetFrontendLoggersList() {
```

Comment:
This function returns a non-const version of member variable. Is this  intended?